File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadPropertyListValueResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;
import org.eclipse.milo.opcua.stack.core.types.builtin.StatusCode;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.HashMap;
import java.util.Map;

public class ReadPropertyListValueResponse {
    private final Map<UaChildId, DataValue> results;

    public ReadPropertyListValueResponse()
    {
        results = new HashMap<>();
    }

    public void setValue(String propertyId, Variant value)
    {
        results.put(new UaChildId(propertyId), new DataValue(value, StatusCode.GOOD, null, null));
    }

    public void setValue(UaChildId subPropertyId, Variant value)
    {
        results.put(subPropertyId, new DataValue(value, StatusCode.GOOD, null, null));
    }

    public void setError(String propertyId, StatusCode errorCode)
    {
        results.put(new UaChildId(propertyId), new DataValue(Variant.NULL_VALUE, errorCode, null, null));
    }

    public void setError(UaChildId subPropertyId, StatusCode errorCode)
    {
        results.put(subPropertyId, new DataValue(Variant.NULL_VALUE, errorCode, null, null));
    }

    public Map<UaChildId, DataValue> getResults() {
        return results;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/ReadPropertyListValueResponse.ts
import { StatusCodes } from "opcua-webapi";
import { makeUaStatusCode, UaDataValue, UaStatusCode, UaVariant } from "opcua-webapi-ts";
import { UaChildId } from "../../../types/common/UaChildId";

export class ReadPropertyListValueResponse {
    private readonly _results: Map<string, UaDataValue>;

    constructor(results: Map<string, UaDataValue> = new Map()) {
        this._results = results;
    }

    setValue(propertyId: string | UaChildId, value: UaVariant): void {
        this._results.set(
            this.toKey(propertyId),
            new UaDataValue(value, makeUaStatusCode(StatusCodes.Good)),
        );
    }

    setError(propertyId: string | UaChildId, errorCode: UaStatusCode): void {
        this._results.set(
            this.toKey(propertyId),
            new UaDataValue(UaVariant.null(), errorCode),
        );
    }

    get results(): Map<string, UaDataValue> {
        return this._results;
    }

    private toKey(propertyId: string | UaChildId): string {
        if (typeof propertyId === "string") {
            return propertyId;
        }

        return propertyId.toString();
    }
}